### ISSUE TITLE
Automatic GROUPR processing for CINDER

### ISSUE THREAD

**Issue by jchsublet** [njoy/NJOY2016#253]
NJOY2016/GROUPR /10 automatic processing from MF=8 dictionary for CINDER encounters some difficulties when it needs to handle MF=10 channel IZAPs when Z is above 99: 1000xZ + A. The below input illustrates the error: 

outFm247: ***error in getsig*** can't find mf,mt,izar,lfs =        10        4      247    0

when in fact the MF10 MT=4 IZAP is given in the evaluation as 100247 ! it needs to read, handled as I6, printed I8 with a LFS if needed

[inFm247.txt](https://github.com/njoy/NJOY2016/files/9149588/inFm247.txt)
[n-Fm257.tendl.txt](https://github.com/njoy/NJOY2016/files/9149591/n-Fm257.tendl.txt)

Having said that more complex mf3*mf6 mt 5 zam targets and products are properly processed but in the printout when Z is also above 99

 for mf3 mf6 mt  5 zam  992542 (n,x) 99254m1 production
 for mf3 mf6 mt  5 zam 1002420 (n,x) ***** production

**Comment by whaeck** [comment]
This should not be that difficult to correct, I think.

However, the evaluation that you have given here is not for Fm247, it is for Fm257 (I got confused when I tried to find MF10 MT4 in that file and it is not defined).

**Comment by jchsublet** [comment]
Of course, sorry here it is [n-Fm247.tendl.txt](https://github.com/njoy/NJOY2016/files/9156647/n-Fm247.tendl.txt)

**Comment by whaeck** [comment]
The issue seem to be related to line 655 in groupr.f90:
```
         izar=(mfd-((mfd/1000000)*1000000))/10
```
For MF10, the mfd valueis set to the IZA value times 10, so for Fm247 this would be 1002470. The above line would truncate that back to 247. Instead of 1e+6, the above equation should use 1e+7. In the code around this line, 1e+7 is used all the time so I assume this is a typo. It worked so far since we rarely work with elements with a Z above 100.

I made a quick fix in fix/groupr-mf10. NJOY now runs to completion on this case.

**Comment by jchsublet** [comment]
I have tested fix/groupr-mf10 and it runs to completion for all 82 neutron induced cases where it did not before, nice job. However to wrap this properly one would also fix the printout of the product, also 1002470 and 100247m1 in the below particular cases. Outgoing also in i6, not i5 as may be in line 1447,1448
 
group constants at t=2.936E+02 deg k                                     1.0s
 for mf10 mt  4 zam 1002470 (n,inel) ***** production                                   
 --- 
group constants at t=2.936E+02 deg k                                     1.1s
 for mf10 mt  4 zam 1002471 (n,inel) *****m1 production

**Comment by whaeck** [comment]
I have increased the output for an iza to i6 from i5 so that fixes this last print out issue.

**Comment by jchsublet** [comment]
This fix/groupr-mf10 works marvellously on all targets, the one that failed previously and all others as before. This clear the path for Z above 99 application library for CINDER. The NJOY2016 groupr output file is also fully functional, listing all mf3, mf10, mf3*mf6 mts channels and residuals
